File: app/src/debug/java/com/bopr/android/smailer/ui/DebugActionProvider.java
package com.bopr.android.smailer.ui;

import android.content.Context;
import android.content.Intent;
import android.support.v4.view.ActionProvider;
import android.view.View;

/**
 * Class DebugActionProvider.
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
@SuppressWarnings("unused") /* used in menu_main.xml */
public class DebugActionProvider extends ActionProvider {

    /**
     * Creates a new instance.
     *
     * @param context Context for accessing resources.
     */
    public DebugActionProvider(Context context) {
        super(context);
    }

    @Override
    public View onCreateActionView() {
        return null;
    }

    @Override
    public boolean onPerformDefaultAction() {
        getContext().startActivity(new Intent(getContext(), DebugActivity.class));
        return true;
    }

}

File: app/src/debug/kotlin/com/bopr/android/smailer/ui/DebugActionProvider.kt
package com.bopr.android.smailer.ui

import android.content.Context
import android.content.Intent
import android.view.View
import androidx.core.view.ActionProvider

/**
 * Used in debug menu_main.xml.
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
class DebugActionProvider (context: Context) : ActionProvider(context) {

    override fun onCreateActionView(): View {
        throw UnsupportedOperationException("DebugActionProvider does ot provide views.")
    }

    override fun onPerformDefaultAction(): Boolean {
        context.startActivity(Intent(context, DebugActivity::class.java))
        return true
    }
}